src/models/CommonModel.ts

```diff
@@ -359,37 +385,47 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * 
    * @param mergeTo 
    * @param mergeFrom 
    * @param originalSchema 
    * @param alreadyIteratedModels
    */
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   private static mergeItems(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema, alreadyIteratedModels: Map<CommonModel, CommonModel> = new Map()) {
-    const merge = (models: CommonModel | CommonModel[] | undefined): CommonModel | undefined => {
-      if (!Array.isArray(models)) {return models;}
-      let mergedItemsModel: CommonModel | undefined = undefined;
-      models.forEach((model, index) => { 
-        Logger.warn(`Found duplicate items at index ${index} for model. Model item for ${mergeFrom.$id || 'unknown'} merged into ${mergeTo.$id || 'unknown'}`, mergeTo, mergeFrom, originalSchema);
-        mergedItemsModel = CommonModel.mergeCommonModels(mergedItemsModel, model, originalSchema, alreadyIteratedModels); 
-      });
-      return mergedItemsModel;
-    };
-    if (mergeFrom.items !== undefined) {
-      //Incase of arrays, merge them into a single model
-      const mergeFromItemsModel = merge(mergeFrom.items);
-      const mergeToItemsModel = merge(mergeTo.items);
-      if (mergeFromItemsModel !== undefined) {
-        if (mergeToItemsModel === undefined) {
-          mergeTo.items = mergeFromItemsModel;
-        } else {
-          Logger.warn(`Found duplicate item for model. Model item for ${mergeFrom.$id || 'unknown'} merged into ${mergeTo.$id || 'unknown'}`, mergeTo, mergeFrom, originalSchema);
-          mergeTo.items = CommonModel.mergeCommonModels(mergeToItemsModel, mergeFromItemsModel, originalSchema, alreadyIteratedModels);
-        }
+    if (mergeFrom.items === undefined) { return; }
+    if (Array.isArray(mergeFrom.items) && mergeFrom.items.length === 0) { return; }
+    if (mergeTo.items === undefined) {
+      mergeTo.items = mergeFrom.items;
+      return;
+    }
+    const mergeToItems = mergeTo.items;
+
+    //mergeFrom and mergeTo is not tuple
+    if (!Array.isArray(mergeFrom.items) && !Array.isArray(mergeToItems)) {
+      mergeTo.items = CommonModel.mergeCommonModels(mergeToItems, mergeFrom.items, originalSchema, alreadyIteratedModels); 
+    }
+
+    //mergeFrom and mergeTo is tuple
+    if (Array.isArray(mergeFrom.items) && Array.isArray(mergeToItems)) {
+      for (const [index, mergeFromTupleModel] of mergeFrom.items.entries()) {
+        (mergeTo.items as CommonModel[])[Number(index)] = CommonModel.mergeCommonModels(mergeToItems[Number(index)], mergeFromTupleModel, originalSchema, alreadyIteratedModels); 
+      }
+    }
+
+    //mergeFrom is a tuple && mergeTo is not
+    if (Array.isArray(mergeFrom.items) && !Array.isArray(mergeToItems)) {
+      mergeTo.items = [];
+      for (const [index, mergeFromTupleModel] of mergeFrom.items.entries()) {
+        mergeTo.items[Number(index)] = CommonModel.mergeCommonModels(mergeToItems, mergeFromTupleModel, originalSchema, alreadyIteratedModels); 
+      }
+    }
```

Comment:
You can merge these ifs to single one `if (Array.isArray(mergeFrom.items)) { if(Array.isArray(mergeToItems)) {} else ... }`

Reply:
I tried to do that, but I feel like it the maintainability declines and it is harder to grasp what is happening 🤷 

Do you think it is better to merge them?

Reply:
The only problem for me is constantly checking one condition, but i can see that it is now easier to understand, with multiple ifs